hostinit/setup.go

```diff
@@ -95,27 +96,40 @@ func (init *HostInit) startHosts(ctx context.Context) error {
 
 		cloudManager, err := providers.GetCloudManager(h.Provider, init.Settings)
 		if err != nil {
-			return errors.Wrapf(err, "error getting cloud provider for host %s", h.Id)
+			grip.Warning(errors.Wrapf(err, "error getting cloud provider for host %s", h.Id))
+			continue
 		}
 
 		err = h.Remove()
 		if err != nil {
-			return errors.Wrapf(err, "error removing intent host %s", h.Id)
+			grip.Notice(errors.Wrapf(err, "error removing intent host %s", h.Id))
+			continue
 		}
 
 		_, err = cloudManager.SpawnHost(&h)
 		if err != nil {
-			return errors.Wrapf(err, "error spawning host %s", h.Id)
+			// we should maybe try and continue-on-error
+			// here, if we get many errors, but the chance
+			// is that if one fails, the chances of others
+			// failing is quite high (at least while all
+			// cloud providers are typically the same
+			// service provider.)
+			err = errors.Wrapf(err, "error spawning host %s", h.Id)
```

Comment:
Presently, AWS is the only fully functional/supported provider right? Otherwise, I'd prefer to to implement something simple here that keeps track of failures on a provider level. 

I'm just thinking a map of whatever type h.Provider is to ints. If the count for the provider is gt 0, skip that host.

Reply:
AWS is the only supported provider in use in our deployment of evergreen.

The reasons for failure are basically: 
- aws is slow or network constrained.
- aws is rate limiting us. 

The truth is, though, that the host intent documents are getting removed (which means that in the event of a spawn failure, we're less likely to end up with a lot of stale hosts. 

I think being smarter here isn't a terrible idea, but at the same time, I think I'd like to ease into more wide-ranging changes.